src/vcf/record.cpp

```diff
@@ -620,28 +661,50 @@ namespace ebi
         return valid;
     }
 
-    void Record::check_field_cardinality(std::string const & field,
-                                         std::vector<std::string> const & values,
-                                         std::string const & number) const
+    void Record::check_info_field_cardinality(std::string const &field, std::vector<std::string> const &values,
+                                              std::string const &number) const
+    {
+        bool valid;
+        long cardinality;
+        size_t ploidy = 0;
+        if (number != G) {
+            check_sample_field_cardinality(field, values, number, ploidy, valid, cardinality);
+        } else {
+            // this case is not well defined, we just skip this check and allow any cardinality
+            // for further discussion see https://github.com/samtools/hts-specs/issues/272
+        }
+    }
+
+    void Record::check_sample_field_cardinality(std::string const &field, std::vector<std::string> const &values,
+                                                std::string const &number, size_t ploidy, bool &valid,
+                                                long &cardinality) const
     {
-        long expected;
-        if(not is_valid_cardinality(number, alternate_alleles.size(), expected)) {
-            raise(std::make_shared<Error>(line, field + " meta specification Number=" + number + " is not one of [A, R, G, ., <non-negative number>]"));
+        valid = is_valid_cardinality(number, alternate_alleles.size(), ploidy, cardinality);
```

Comment:
`valid` doesn't seem to be used by any of the methods that call this one, so it could be a local variable. In my opinion the method call was clear enough, because it expressed what the validity applied to (cardinality), whereas the variable name is slightly more ambiguous.

Reply:
it's used in check_sample_subfields_cardinality_type

Reply:
I don't think we actually needed that parameter, so I could revert this back